pandas/core/indexes/interval.py

```diff
@@ -727,7 +727,14 @@ def _get_indexer_pointwise(
                 if isinstance(locs, slice):
                     # Only needed for get_indexer_non_unique
                     locs = np.arange(locs.start, locs.stop, locs.step, dtype="intp")
-                locs = np.array(locs, ndmin=1)
+                elif lib.is_integer(locs):
+                    locs = np.array(locs, ndmin=1)
+                else:
+                    # FIXME: This is wrong; its boolean; not reached
+                    # error: Item "int" of "Union[int, ndarray[Any, Any]]"
+                    #  has no attribute "dtype"
+                    assert locs.dtype.kind == "i"  # type: ignore[union-attr]
```

Comment:
How did you run into this? (don't directly see the relation with this PR)

Reply:
i think this was a note to self when i realized that the then-current code was wrong.  fortunately it has been fixed on an unrelated PR since then.